### PR TITLE
plugins: add dotnet plugin (CRAFT-612)

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

-------

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: tests/integration/plugins/test_dotnet.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from pathlib import Path
+
+import pytest
+import yaml
+
+from craft_parts import LifecycleManager, Step
+
+
+@pytest.mark.usefixtures("mocker")
+def test_go_plugin(new_dir):
```
Comment (cmatsuoka) [inline tests/integration/plugins/test_dotnet.py:28]: This should be dotnet.

File: tests/integration/plugins/test_dotnet.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import textwrap
+from pathlib import Path
+
+import pytest
+import yaml
+
+from craft_parts import LifecycleManager, Step
+
+
+@pytest.mark.usefixtures("mocker")
```
Comment (cmatsuoka) [inline tests/integration/plugins/test_dotnet.py:27]: Where is mocker used in this test?

Comment (sergiusens) [reply tests/integration/plugins/test_dotnet.py:27]: I copied that from the go plugin one :-P

**Review by cmatsuoka — APPROVED** [review]
Thanks!

**Pull request merged**